pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep_test.go

```diff
@@ -0,0 +1,94 @@
+package preaction
+
+import (
+	"context"
+	pmock "github.com/kyma-incubator/reconciler/pkg/reconciler/chart/mocks"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes/mocks"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-incubator/reconciler/pkg/logger"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+)
+
+const (
+	manifestString = "testManifest"
+	//kyma1xVersion  = "1.24.8"
+	//kyma2xVersion  = "2.0"
+)
+
+func TestDeletingNatsOperatorResources(t *testing.T) {
+	action, actionContext, mockProvider, k8sClient, mockedComponentBuilder := testSetup()
+
+	// execute the step
+	err := action.Execute(actionContext, actionContext.Logger)
+	require.NoError(t, err)
+
+	// ensure the right calls were invoked
+	mockProvider.AssertCalled(t, "RenderManifest", mockedComponentBuilder)
+	k8sClient.AssertCalled(t, "Delete", actionContext.Context, manifestString, namespace)
+	k8sClient.AssertCalled(t, "DeleteResource", crdPlural, natsOperatorCRDsToDelete[0], namespace)
+	k8sClient.AssertCalled(t, "DeleteResource", crdPlural, natsOperatorCRDsToDelete[1], namespace)
+}
+
+// todo execute this test, when the check for kyma2x version is available, see the the todo comment from removenatsoperatorstep:Execute()
```

Comment:
Please open also an issue for this so that we won't forget it

Reply:
https://github.com/kyma-incubator/reconciler/issues/453